ixdiagnose/artifacts/items/file.py

```diff
@@ -0,0 +1,39 @@
+import os
+import pathlib
+import shutil
+
+from ixdiagnose.utils.io import truncate_file
+from typing import Optional, Tuple
+
+from .base import Item
+
+
+def get_file_size(file_path: str) -> int:
+    return pathlib.Path(file_path).lstat().st_size
+
+
+class File(Item):
```

Comment:
I think it would be better from standpoint of avoiding TOCTOU issues if we opened the files with O_NOFOLLOW and perform the handle-based syscall variants (fstat, ftruncate, etc). Path-based operations like shutil.copy2 can work with an open handle if you pass the path `/proc/self/fd/<fd>` to it.